src/omv/framebuffer.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of the OpenMV project.
+ * Copyright (c) 2013/2014 Ibrahim Abdelkader <i.abdalkader@gmail.com>
+ * This work is licensed under the MIT license, see the file LICENSE for details.
+ *
+ * Framebuffer stuff.
+ *
+ */
+#include "imlib.h"
+#include "omv_boardconfig.h"
+#include "framebuffer.h"
+// If buffer size is bigger than this threshold, the quality is reduced.
+// This is only used for JPEG images sent to the IDE not normal compression.
+#define JPEG_QUALITY_THRESH     (160*120*2)
+
+extern char _fb_base;
+framebuffer_t *fb = (framebuffer_t *) &_fb_base;
```

Comment:
There's a problem with allocating the buffer this way, it's global, can't have other fbs or jpeg_fbs.

EDIT1: And if you make it static we'll have to use getters fb_get_main_fb() and fb_get_jpeg_fb(). could add those functions and #define MAIN_FB() fb_get_main_fb()

EDIT2: So if you add getters and #define the FB macros, this way it won't break the code and it gets refactored to use the getters. I suggest **fb_get_raw/main_buffer()** and **fb_get_jpeg_buffer()**